core-rust/state-manager/src/staging/mod.rs

```diff
@@ -63,6 +63,43 @@
  */
 
 mod cache;
+mod result;
 mod stage_tree;
 
+use crate::accumulator_tree::storage::ReadableAccuTreeStore;
+use crate::{ReceiptTreeHash, TransactionTreeHash};
+use radix_engine::ledger::ReadableSubstateStore;
+use radix_engine_interface::api::types::SubstateOffset;
+use radix_engine_stores::hash_tree::tree_store::{ReNodeModulePayload, ReadableTreeStore};
+
 pub use cache::*;
+pub use result::*;
+
+pub trait ReadableLayeredTreeStore:
+    ReadableTreeStore<ReNodeModulePayload> + ReadableTreeStore<SubstateOffset>
+{
+}
+impl<T: ReadableTreeStore<ReNodeModulePayload> + ReadableTreeStore<SubstateOffset>>
+    ReadableLayeredTreeStore for T
+{
+}
+
+pub trait ReadableTxAndRxTreeStore:
+    ReadableAccuTreeStore<u64, TransactionTreeHash> + ReadableAccuTreeStore<u64, ReceiptTreeHash>
+{
+}
+impl<
+        T: ReadableAccuTreeStore<u64, TransactionTreeHash>
```

Comment:
What does Accu stand for? Could we maybe give it a clearer name?

Reply:
let's discuss only at https://github.com/radixdlt/babylon-node/pull/357#discussion_r1137227051